pkg/controller/cloudantv1/cloudantdatabase.go

```diff
@@ -0,0 +1,213 @@
+		managed.WithLogger(log),
+		managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name))))
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		For(&v1alpha1.CloudantDatabase{}).
+		Complete(r)
+}
+
+// A cloudantdatabaseConnector is expected to produce an ExternalClient when its Connect method
+// is called.
+type cloudantdatabaseConnector struct {
+	kube     client.Client
+	usage    resource.Tracker
+	clientFn func(optd ibmc.ClientOptions) (ibmc.ClientSession, error)
+	logger   logging.Logger
+}
+
+// Connect produces an ExternalClient for IBM Cloud API
+func (c *cloudantdatabaseConnector) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	opts, err := ibmc.GetAuthInfo(ctx, c.kube, mg)
+	if err != nil {
+		return nil, errors.Wrap(err, ibmc.ErrGetAuth)
+	}
+
+	service, err := c.clientFn(opts)
+	if err != nil {
+		return nil, errors.Wrap(err, ibmc.ErrNewClient)
+	}
+
+	return &cloudantdatabaseExternal{client: service, kube: c.kube, logger: c.logger}, nil
+}
+
+// An cloudantdatabaseExternal observes, then either creates, updates, or deletes an
+// external resource to ensure it reflects the managed resource's desired state.
+type cloudantdatabaseExternal struct {
+	client ibmc.ClientSession
+	kube   client.Client
+	logger logging.Logger
+}
+
+func (c *cloudantdatabaseExternal) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) { // nolint:gocyclo
+	cr, ok := mg.(*v1alpha1.CloudantDatabase)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errNotCloudantDatabase)
+	}
+
+	if meta.GetExternalName(cr) == "" {
+		return managed.ExternalObservation{
+			ResourceExists: false,
+		}, nil
+	}
+
+	instance, _, err := c.client.CloudantV1().GetDatabaseInformation(&cv1.GetDatabaseInformationOptions{Db: reference.ToPtrValue(meta.GetExternalName(cr))})
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(ibmc.IsResourceNotFound, err), errGetCloudantDatabaseFailed)
+	}
+
+	// is updateSeq the correct thing to use for state ??
+	// is there anything returned in databaseinformation that could be used for state or 
+	// should I define state similar to the way I did for topic
+	// I don't know what the options for updateseq are right now so I'm leaving "active",
+	// "inactive", and "provisioning", ??
+	if !(reference.FromPtrValue(instance.UpdateSeq) == "active" ||
+		reference.FromPtrValue(instance.UpdateSeq) == "inactive" ||
+		reference.FromPtrValue(instance.UpdateSeq) == "provisioning") {
+		return managed.ExternalObservation{ResourceExists: false}, nil
+	}
+
+	currentSpec := cr.Spec.ForProvider.DeepCopy()
+	if err = ibmccdb.LateInitializeSpec(&cr.Spec.ForProvider, instance); err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, ibmc.ErrManagedUpdateFailed)
+	}
+	if !cmp.Equal(currentSpec, &cr.Spec.ForProvider) {
+		// if there is no update what should be called here ??
```

Comment:
This is a different update - it is updating the Specs based on LateInitializeSpec() to fill in specs that might be not filled in by the user but are retrieved from the observation (if any). So it's OK the way you have it now